packages/aws-client/src/DynamoDbSearchQueue.ts

```diff
@@ -28,30 +35,24 @@ class DynamoDbSearchQueue {
    *
    * When there are no more items in the queue, returns 'null'.
    *
-   * @returns {Promise<Object>} - an item from the DynamoDB table
+   * @returns {Promise<Object>} an item from the DynamoDB table
    */
   async shift() {
     if (this.items.length === 0) await this.fetchItems();
     return this.items.shift();
   }
 
-  /**
-   * Query the DynamoDB API to get the next batch of items
-   *
-   * @returns {Promise<undefined>} - resolves when the queue has been updated
-   * @private
-   */
   async fetchItems() {
     let response;
     do {
       response = await this.dynamodbDocClient[this.searchType](this.params).promise(); // eslint-disable-line no-await-in-loop, max-len
       if (response.LastEvaluatedKey) this.params.ExclusiveStartKey = response.LastEvaluatedKey;
-    } while (response.Items.length === 0 && response.LastEvaluatedKey);
+    } while ((response.Items || []).length === 0 && response.LastEvaluatedKey);
```

Comment:
I don't quite understand the refactor

Reply:
Apparently `response.Items` can potentially be `undefined`. This catches that case and defaults to an empty array, which gives the behavior that we're looking for.